### PR TITLE
WIP:Kyle feature import staff bookings tool http objects

### PR DESCRIPTION
**Pull request by kylebarney**
Tracking work in progress here so that we don't duplicate any efforts. **_Do not merge this yet._**

The first commit has the imported Http objects that we're moving into this repo from the `staff-booking` tool. I'm working on getting the imported files all converted and working properly.

### CODE REVIEW

**Comment by sl0wik** [comment]
@kylebarney If it should not be merged always add "WIP: " prefix.

File: routes/api.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+use Illuminate\Support\Facades\Route;
+
+Route::post('/{location:slug}/checkout', [Tipoff\Booking\Http\Controllers\CheckoutController::class, 'store']);
+Route::get('/locations/{location:slug}', [Tipoff\Booking\Http\Controllers\LocationsController::class, 'show']);
+Route::post('/{location:slug}/contact-details', [Tipoff\Booking\Http\Controllers\ContactDetailsController::class, 'store']);
+Route::post('/{user:id}/voucher-applications', [Tipoff\Booking\Http\Controllers\VoucherApplicationsController::class, 'store']);
```
Comment (sl0wik) [inline routes/api.php:8]: Is `:id` required? As far I know resolver would use https://github.com/illuminate/database/blob/master/Eloquent/Model.php#L1706 by default so such syntax should be used when model is resolved by a non-default key.

Comment (kylebarney) [reply routes/api.php:8]: Correct, this is just copied and pasted from the existing code so far, so I haven't changed anything. But good catch.

**Comment by pdbreen** [comment]
@drewroberts  This PR should be killed. This staff-bookings should not become a dependency of this package or be integrated with this package.  It depends on checkout which bookings cannot depend on.

**Comment by pdbreen** [comment]
Instead, the staff-bookings package pulls in both bookings and checkout.

**Comment by pdbreen** [comment]
Everything in this PR is included in this PR  - https://github.com/tipoff/staff-bookings/pull/23 - as well as being updated for current Cart/CartItem interface support.

**Comment by drewroberts** [comment]
Thank you.

**Pull request closed without merging**